### PR TITLE
Run tensorflow & keras autologging tests faster

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

The `tensorflow` check roughly takes **30~35 minutes** (`Run tests` job):

https://github.com/mlflow/mlflow/runs/1554207916

This PR refactors the following files to run the check faster:

- `test_keras_autolog.py`
- `test_tensorlfow_autolog.py`
- `test_tensorlflow_autolog.py`

## How is this patch tested?

Fixed unit tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/keras_autolog/test_keras_autolog.py [hunk 1]
```diff
@@ -14,12 +14,12 @@
 
 @pytest.fixture
 def random_train_data():
-    return np.random.random((1000, 32))
+    return np.random.random((150, 4))
```
Comment (harupy) [inline tests/keras_autolog/test_keras_autolog.py:17]: reduced the size of features

File: tests/keras_autolog/test_keras_autolog.py [hunk 2]
```diff
@@ -14,12 +14,12 @@
...
 
 
 @pytest.fixture
 def random_one_hot_labels():
-    n, n_class = (1000, 10)
+    n, n_class = (150, 3)
```
Comment (harupy) [inline tests/keras_autolog/test_keras_autolog.py:22]: reduced the number of classes

File: tests/keras_autolog/test_keras_autolog.py [hunk 3]
```diff
@@ -37,9 +37,8 @@ def manual_run(request):
 def create_model():
     model = keras.Sequential()
 
-    model.add(layers.Dense(64, activation="relu", input_shape=(32,)))
-    model.add(layers.Dense(64, activation="relu"))
-    model.add(layers.Dense(10, activation="softmax"))
+    model.add(layers.Dense(32, activation="relu", input_shape=(4,)))
+    model.add(layers.Dense(3, activation="softmax"))
```
Comment (harupy) [inline tests/keras_autolog/test_keras_autolog.py:41]: Simplified the model structure

**Comment by harupy** [comment]
@dbczumar I just noticed updating files under `tests/<flavor_name>_autlo/` doesn't trigger the cross version tests. This is because `set_matrix.py` expects this format: `tests/<flavor_name>/`.

**Comment by harupy** [comment]
Looks like I need to merge #3844 first.

File: tests/keras_autolog/test_keras_autolog.py (hunk 1 above)
Comment (harupy) [reply tests/keras_autolog/test_keras_autolog.py:17]: the same shape as the iris dataset

**Comment by harupy** [comment]
@dbczumar Found a magic here: https://stackoverflow.com/questions/52725772/keras-why-is-load-model-getting-slower-each-time

**Comment by dbczumar** [comment]
> @dbczumar Found some magic here: https://stackoverflow.com/questions/52725772/keras-why-is-load-model-getting-slower-each-time

Wow! That’s really neat / concerning! Nice find!

**Comment by harupy** [comment]
It's 15 min. :)

File: tests/keras_autolog/test_keras_autolog.py [hunk 4]
```diff
@@ -12,14 +12,22 @@
 from unittest.mock import patch  # noqa
 
 
+@pytest.fixture(autouse=True)
+def clear_session():
+    yield
+    # Release the global state managed by Keras to avoid this issue:
+    # https://github.com/keras-team/keras/issues/2102
+    keras.backend.clear_session()
```
Comment (harupy) [inline tests/keras_autolog/test_keras_autolog.py:20]: This issue occurs in old versions (< 2.3) of keras:

In the cross version tests,

- test (keras / 2.3.1 / autologging): took **3m 27s**
- test (keras / 2.2.5 / autologging): took **16m 9s**

https://github.com/mlflow/mlflow/runs/1562184382?check_suite_focus=true

**Review by dbczumar — APPROVED** [review]
LGTM! this is awesome!

**Pull request merged**